src/ORM/Connect/MySQLiConnector.php

```diff
@@ -65,12 +66,19 @@ public function prepareStatement($sql, &$success)
         // Record last statement for error reporting
         $statement = $this->dbConn->stmt_init();
         $this->setLastStatement($statement);
+
         try {
             $success = $statement->prepare($sql);
         } catch (mysqli_sql_exception $e) {
             $success = false;
-            $this->databaseError($e->getMessage(), E_USER_ERROR, $sql);
+            $this->throwRelevantError($e->getMessage(), $e->getCode(), E_USER_ERROR, $sql, []);
+        }
+
+        if (!$success || $statement->error) {
+            $this->throwRelevantError($this->getLastError(), $this->getLastErrorCode(), E_USER_ERROR, $sql, []);
+            return null;
         }
```

Comment:
Two things to note here:

1. The new exception won't ever get thrown from here, but I want to use the new method for consistency and to allow future specific exceptions to be thrown which _might_ be relevant here.
2. The new `if (!$success || $statement->error) {` condition is required to pass the new tests for when report mode is set to `MYSQLI_REPORT_OFF` - this makes it consistent with everywhere else we're throwing these.